### ISSUE TITLE
Error running model with multi layer ONNX local functions

### ISSUE THREAD

**Issue by BowenBao** [microsoft/onnxruntime#16813]
Update: the latest model can be downloaded here https://microsoft-my.sharepoint.com/:u:/p/bowbao/EVWNeJy8v4ZOg9uN_q_ca_QBy9PaYAGYmI8GOPm_XdNOaA?e=tMa379

Transpose node (within aten_unfold) is fixed in the latest model.

After the fix, only 1 model fails during run.

### Describe the issue

This model is produced by latest `torch.onnx.dynamo_export` w/ PyTorch 2. ONNX local function is used heavily in the model to preserve structural information. 

Sometimes ORT emits errors running the model when it has many layers of functions, but passes after the model is being inlined by `onnx.inliner`. 

This is model `eca_halonext26ts` from torchbench. Four example onnx models with different level of onnx local function adoption is provided.
All model passes onnx shape inference and checker. Only one model runs in ORT successfully.

### To reproduce

Model can be [~~downloaded from here~~](https://microsoft-my.sharepoint.com/:u:/p/bowbao/Eev5htIKOkNHh4NDaF2c6e8BXLTSoI3A2hksQbgqFY7wlw?e=Hi563k)  [updated version](https://microsoft-my.sharepoint.com/:u:/p/bowbao/EVWNeJy8v4ZOg9uN_q_ca_QBy9PaYAGYmI8GOPm_XdNOaA?e=tMa379)

```python
import onnxruntime
import numpy as np
import onnx

sample_input = np.random.rand(8, 3, 256, 256).astype(np.float32)

model_names = [
    "eca_halonext26ts_nomod.onnx",  # Model w/o exporting nn.Module as local function
    "eca_halonext26ts_inline_nomod.onnx",  # above model inlined by onnx.inliner
    "eca_halonext26ts_mod.onnx",  # Model w/ exporting nn.Module as local function
    "eca_halonext26ts_inline_mod.onnx",  # above model inlined by onnx.inliner
]

for model_name in model_names:
    model = onnx.load(model_name)
    onnx.shape_inference.infer_shapes(model, check_type=True, strict_mode=True, data_prop=True)
    onnx.checker.check_model(model, full_check=True)

    try:
        session_options = onnxruntime.SessionOptions()
        session_options.log_severity_level = 4
        session = onnxruntime.InferenceSession(model_name, sess_options=session_options, providers=["CPUExecutionProvider"])
        session.run(None, {"arg0": sample_input})
    except Exception as e:
        print(f"========== {model_name} failed: {e}")
    else:
        print(f"========== {model_name} passed")

```

> ========== eca_halonext26ts_nomod.onnx passed

>========== eca_halonext26ts_inline_nomod.onnx passed

>========== eca_halonext26ts_mod.onnx failed: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Exception during initialization: /onnxruntime_src/onnxruntime/core/graph/graph.cc:3538 void onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs(const std::unordered_set<std::basic_string<char> >*) initializer_node_arg != nullptr was false. Cannot find NodeArgs for [_inline_timm_models_layers_norm_act_BatchNormAct2d_stem_conv1_bn_1_val_5]

>========== eca_halonext26ts_inline_mod.onnx passed

### Urgency

eca_halonext26ts from [onnx export bench](https://github.com/microsoft/onnx-converters-private/issues/166)

### Platform

Linux

### OS Version

Ubuntu 20.04

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

ort-nightly                   1.16.0.dev20230711005

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

Default CPU

**Comment by xadupre** [comment]
adding @justinchuby

**Comment by yuslepukhin** [comment]
I tried the first model with the script with the tip of main. The onnx inference throws an exception and never completes the script.
Barring that, native ORT inference continues for a long time and adds lots of allocations when calling into onnx inference code.
I am seeing lots of fixes in onnx inference code. Will see if that makes any difference.

**Comment by BowenBao** [comment]
@yuslepukhin thanks for investigating! The issue you discovered about transpose dim size and rank is indeed a bug within torchlib, which we have triaged and fixed. I'm keeping this issue open to track the behavioral difference between models with different number of layers of local functions.

**Comment by yuslepukhin** [comment]
The shape inference actually throws when the input dims are known to mismatch perms rank at inference time. But at runtime, the model can feed a wrong number of dims.

**Comment by yuslepukhin** [comment]
I have checked all of the models. One of them `eca_halonext26ts_mod.onnx` fails Constant node shape inference, that requires one of the following attributes to be present, so it does not progress very far.

>   auto* value = ctx.getAttribute("value");
  auto* sparse_value = ctx.getAttribute("sparse_value");
  auto* value_int = ctx.getAttribute("value_int");
  auto* value_ints = ctx.getAttribute("value_ints");
  auto* value_float = ctx.getAttribute("value_float");
  auto* value_floats = ctx.getAttribute("value_floats");
  auto* value_string = ctx.getAttribute("value_string");
  auto* value_strings = ctx.getAttribute("value_strings");

**Comment by yuslepukhin** [comment]
All other models fail due to the `transpose` perm vs input rank mismatch. If you provide updates, I will try to run it further.

**Comment by pranavsharma** [comment]
Can this be closed now?
